AEPMessaging/Sources/MessagingPropositionInteraction.swift

```diff
@@ -0,0 +1,111 @@
+        ]
+
+        if
+            let itemId = itemId,
+            !itemId.isEmpty
+        {
+            propositionDetailsData[MessagingConstants.XDM.Inbound.Key.ITEMS] = [
+                [
+                    MessagingConstants.XDM.Inbound.Key.ID: itemId
+                ]
+            ]
+        }
+
+        let propositionEventType: [String: Any] = [
+            eventType.propositionEventType: 1
+        ]
+
+        var decisioning: [String: Any] = [
+            MessagingConstants.XDM.Inbound.Key.PROPOSITION_EVENT_TYPE: propositionEventType,
+            MessagingConstants.XDM.Inbound.Key.PROPOSITIONS: [propositionDetailsData]
+        ]
+
+        // only add `propositionAction` data if this is an interact event
+        if
+            eventType == .interact,
+            let interaction = interaction
+        {
+            let propositionAction: [String: String] = [
+                MessagingConstants.XDM.Inbound.Key.ID: interaction,
+                MessagingConstants.XDM.Inbound.Key.LABEL: interaction
+            ]
+            decisioning[MessagingConstants.XDM.Inbound.Key.PROPOSITION_ACTION] = propositionAction
+        }
+
+        let experience: [String: Any] = [
+            MessagingConstants.XDM.Inbound.Key.DECISIONING: decisioning
+        ]
+
+        let xdm: [String: Any] = [
+            MessagingConstants.XDM.Key.EVENT_TYPE: eventType.toString(),
+            MessagingConstants.XDM.AdobeKeys.EXPERIENCE: experience
+        ]
+        return xdm
+    }
+
+    enum CodingKeys: String, CodingKey {
+        case eventType
+        case interaction
+        case propositionInfo
+        case itemId
+    }
+
+    init(eventType: MessagingEdgeEventType, interaction: String?, propositionInfo: PropositionInfo, itemId: String?) {
+        self.eventType = eventType
+        self.interaction = interaction
+        self.propositionInfo = propositionInfo
+        self.itemId = itemId
+    }
+
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encode(eventType.toString(), forKey: .eventType)
+        try container.encode(interaction, forKey: .interaction)
+        try container.encode(propositionInfo, forKey: .propositionInfo)
+        try container.encode(itemId, forKey: .itemId)
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        guard
+            let eventTypeString = try? container.decode(String.self, forKey: .eventType),
+            let eventType = MessagingEdgeEventType(fromType: eventTypeString)
+        else {
+            throw DecodingError.dataCorruptedError(forKey: CodingKeys.eventType, in: container, debugDescription: "Cannot create MessagingEdgeEventType from the provided event type string.")
+        }
+        self.eventType = eventType
```

Comment:
should we use a different name for the local `eventType` to make this code a little cleaner?